### PR TITLE
Add inline array marshaller sample to our custom marshalling sample.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
This won't build successfully until the InlineArray feature is out of preview, as the project file doesn't set `LangVersion` to `Preview`.

Fixes dotnet/runtime#86572

### CODE REVIEW

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Comment by BillWagner** [comment]
Hi @jkoritzinsky 

You can add the `<LangVersion>preview</LangVersion>` element in the csproj. We've got tooling that removes it when a new release goes GA. 

It's one of the tools that helps us keep up to date as we write docs for preview features, and then reset for the next version.

File: core/interop/source-generation/custom-marshalling/src/custommarshalling/custommarshalling.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>net7.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
```
Comment (elinor-fung) [inline core/interop/source-generation/custom-marshalling/src/custommarshalling/custommarshalling.csproj:5]: Update the version listed as a prerequisite in the readme?

**Review by elinor-fung — APPROVED** [review]

**Comment by BillWagner** [comment]
@jkoritzinsky 

> any ideas on how to get this building with a .NET 8 SDK since it needs to build with the VS host (due to the C++ project)?

Add a `snippets5000.json` file, like here: https://github.com/dotnet/samples/blob/main/framework/wcf/Basic/Ajax/ComplexTypeAjaxService/CS/snippets.5000.json

That specifies using Visual Studio as the host.

**Comment by jkoritzinsky** [comment]
I did that, but now it's complaining that there's no .NET 8 SDK available.

**Comment by BillWagner** [comment]
> I did that, but now it's complaining that there's no .NET 8 SDK available.

@jkoritzinsky 

It looks like you're the first to need .NET 8 in this repo.  Update this line with the .NET installer channel: https://github.com/dotnet/samples/blob/main/.github/workflows/build-validation.yml#L18

Then you should be fine.  It should match the channel in the `dotnet/docs` repo: https://github.com/dotnet/docs/blob/main/.github/workflows/snippets5000.yml#L18

**Comment by jkoritzinsky** [comment]
I already did that and I'm still getting the same error. If I don't use the visual studio host, it correctly finds the .NET 8 SDK, but it doesn't seem to work with the VS host.

**Comment by BillWagner** [comment]
adding @adegeo 

I've exhausted my knowledge of how the tooling works. He may know.

**Comment by adegeo** [comment]
I don't think this would work with Visual Studio. Doesn't Visual Studio use it's own copies of .NET? The latest versions of Visual Studio probably don't install .NET 8 preview as part of the base image. You'll need to remove the snippets config file and just use the normal `dotnet` host with the now modified build-validation file that installs .NET 8

**Comment by jkoritzinsky** [comment]
The issue is that the .NET SDK isn't hooked up to find the VC++ targets, so the imports in the C++ project fail.

**Comment by adegeo** [comment]
@jkoritzinsky We may have to just merge without the testing framework as this may just be outside of the realm we support. Question, though, the readme indicates that you can do `dotnet build`, are they supposed to also do something else to get the C++ target?

**Comment by jkoritzinsky** [comment]
If you're in a VS command prompt as the instructions say, the targets will be found.

**Comment by jkoritzinsky** [comment]
@adegeo if you're okay with it, I'm okay merging this without the testing framework. We can also wait until .NET 8 releases, whichever you prefer. Once .NET 8 comes out and the VS version on the testing machines is updated, then the tests will start passing.

**Comment by BillWagner** [comment]
Merging based on preview builds.

**Pull request merged**